modules/videoio/include/opencv2/videoio.hpp

```diff
@@ -852,6 +887,32 @@ class CV_EXPORTS_W VideoCapture
     */
     CV_WRAP virtual bool open(int index, int apiPreference, const std::vector<int>& params);
 
+    /** @brief Opens a video using data stream.
+
+    @overload
+
+    The `params` parameter allows to specify extra parameters encoded as pairs `(paramId_1, paramValue_1, paramId_2, paramValue_2, ...)`.
+    See cv::VideoCaptureProperties
+
+    @return `true` if the file has been successfully opened
+
+    The method first calls VideoCapture::release to close the already opened file or camera.
+     */
+    CV_WRAP virtual bool open(const Ptr<IStreamReader>& source, int apiPreference, const std::vector<int>& params);
+
+    /** @brief Opens a video using data stream.
+
+    @overload
+
+    The `params` parameter allows to specify extra parameters encoded as pairs `(paramId_1, paramValue_1, paramId_2, paramValue_2, ...)`.
+    See cv::VideoCaptureProperties
+
+    @return `true` if the file has been successfully opened
+
+    The method first calls VideoCapture::release to close the already opened file or camera.
+     */
+    virtual bool open(const Ptr<std::streambuf>& source, int apiPreference, const std::vector<int>& params);
```

Comment:
Do we really need methods with `std::streambuf`? Maybe it would be better to make `StreambufReadStream` or similar class or just its factory public instead? For example:
```.cpp
shared_ptr<IStreamReader> source = createStdStream(filename); // can accept filename or &&filebuf or &&istream or &&ifstream
VideoCapture cap(source, ...);
```

Reply:
Good point. But I think we can even remove it completely. `std::streambuf` might introduce more limitations than just seek/read interface.